src/backend/core/models.py

```diff
@@ -750,19 +873,26 @@ def soft_delete(self):
         Soft delete the document, marking the deletion on descendants.
         We still keep the .delete() method untouched for programmatic purposes.
         """
-        if self.deleted_at or self.ancestors_deleted_at:
-            raise RuntimeError(
-                "This document is already deleted or has deleted ancestors."
-            )
-
-        # Check if any ancestors are deleted
-        if self.get_ancestors().filter(deleted_at__isnull=False).exists():
+        if (
+            self._meta.model.objects.filter(
+                models.Q(deleted_at__isnull=False)
+                | models.Q(ancestors_deleted_at__isnull=False),
+                pk=self.pk,
+            ).exists()
+            or self.get_ancestors().filter(deleted_at__isnull=False).exists()
+        ):
             raise RuntimeError(
-                "Cannot delete this document because one or more ancestors are already deleted."
+                _("This document is already deleted or has deleted ancestors.")
```

Comment:
Is this displayed/user in the frontend? I'm not sure de should translate errors, because it makes "grep" in logs quite complex.

Reply:
agreed! These types of messages should not be translated :pray: